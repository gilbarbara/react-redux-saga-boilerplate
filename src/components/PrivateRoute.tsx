import { ReactElement } from 'react';
import { Navigate, useLocation } from 'react-router-dom';

interface Props {
  children: ReactElement;
  isAuthenticated: boolean;
  to?: string;
}

export default function PrivateRoute(props: Props) {
  const { children, isAuthenticated, to = '/login' } = props;

  const { pathname = '' } = useLocation();

  return isAuthenticated ? (
    children
  ) : (
    <Navigate state={{ redirect: pathname, isAuthenticated }} to={to} />
  );
}
