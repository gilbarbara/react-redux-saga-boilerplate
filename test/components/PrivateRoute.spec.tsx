import React from 'react';
import { render } from 'test-utils';

import PrivateRoute from '~/components/PrivateRoute';

describe('PrivateRoute', () => {
  it('should redirect for unauthenticated access', () => {
    const { container } = render(
      <PrivateRoute isAuthenticated={false}>
        <div>PRIVATE</div>
      </PrivateRoute>,
      { path: '/private', pathname: '/private' },
    );

    expect(container).toMatchSnapshot();
  });

  it('should allow navigation for authenticated access', () => {
    const { container } = render(
      <PrivateRoute isAuthenticated>
        <div>PRIVATE</div>
      </PrivateRoute>,
      { path: '/private', pathname: '/private' },
    );

    expect(container).toMatchSnapshot();
  });
});
