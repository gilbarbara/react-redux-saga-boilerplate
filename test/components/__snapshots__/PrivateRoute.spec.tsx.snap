// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PrivateRoute > should allow navigation for authenticated access 1`] = `
<div>
  <div>
    PRIVATE
  </div>
</div>
`;

exports[`PrivateRoute > should redirect for unauthenticated access 1`] = `
<div>
  <div
    data-to="/login"
  >
    NAVIGATE
  </div>
</div>
`;
